packages/react-hooks/src/mixer/useMixer.ts

```diff
@@ -0,0 +1,87 @@
+import { MixerContext, MixerContextData } from '@webb-dapp/react-environment';
+import { useCall } from '@webb-dapp/react-hooks/useCall';
+import { LoggerService } from '@webb-tools/app-util';
+import { Mixer, MixerAssetGroup } from '@webb-tools/sdk-mixer';
+import { GroupId } from '@webb-tools/types/interfaces';
+import { useCallback, useContext, useMemo, useState } from 'react';
+// @ts-ignore
+import Worker from './mixer.worker';
+
+/**
+ * @name useMixer
+ * @description get Mixer context value
+ */
+export const useMixerProvider = (): MixerContextData => {
+  return useContext<MixerContextData>(MixerContext);
+};
+
+export const useMixerGroups = (): MixerAssetGroup[] => {
+  const mixerGroupIds = useCall<Array<GroupId>>('query.mixer.mixerGroupIds', [], undefined, []);
+
+  return useMemo(() => {
+    return (
+      mixerGroupIds?.map((id) => {
+        return new MixerAssetGroup(Number(id.toString()), 'EDG', 32);
+      }) ?? []
+    );
+  }, [mixerGroupIds]);
+};
+
+const logger = LoggerService.new('MixerUsage');
+
+export const useMixer = () => {
+  const mixerAssetGroup = useMixerGroups();
+
+  const [mixerResult, setMixerResult] = useState<Omit<MixerContextData, 'init'>>({
+    initialized: false,
+    loading: false,
+    mixer: null,
+    shouldDestroy: false,
+  });
+
+  const [called, setCalled] = useState(false);
+
+  const init = useCallback(async () => {
+    setCalled(true);
+
+    if (called || mixerResult.initialized) {
+      logger.info(`Generating new mixer`);
+      logger.error(`Mixer already initialize`);
```

Comment:
This seems odd to have one after the other no? It doesn't look like it generates a new mixer but just returns undefined and so the log is a bit confusing.

Reply:
Might be good to remove the `logger.info` call.